cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -15,27 +12,25 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
+
+
   def parse(): Unit = {
 
     println(s"*** File $file")
+    var allocationInNewTlab: Boolean = false
+    var allocationOutsideTlab: Boolean = false
+    var methodProfilingSample: Boolean = false
     try {
-      val recording = FlightRecordingLoader.loadFile(file)
-      for (t <- recording.getEventTypes.asScala) {
-        println(t.getName)
-      }
-      val view = recording.createView
-      view.asScala.find(_.getEventType.getName == "GC TLAB Configuration") match {
-        case Some(e) => handleTlabConfiguration(e)
-        case _ => ???
-      }
-
-      for (event <- view.asScala) {
-        event.getEventType.getName match {
-          case "Allocation in new TLAB" => allocation(event, true)
-          case "Allocation outside TLAB" => allocation(event, false)
-          case "Method Profiling Sample" => cpu(event)
+      val recordingFile = new RecordingFile(file.toPath)
+      //      for (event <- view.asScala) {
+      while (recordingFile.hasMoreEvents){
+        val event = recordingFile.readEvent()
+        event.getEventType.getId match {
+          case 331 => allocationInNewTlab=true; allocation(event, true) // Allocation in New TLAB
```

Comment:
no magic number please - 
is this a stable number ( across Java releases /JVM vendors)
can we refer to a canonical definition of that value